container_opts.go

```diff
@@ -324,3 +326,37 @@ func WithSpec(s *oci.Spec, opts ...oci.SpecOpts) NewContainerOpts {
 func WithoutRefreshedMetadata(i *InfoConfig) {
 	i.Refresh = false
 }
+
+// WithCDI updates OCI spec with CDI content
+func WithCDI(s *oci.Spec, annotations map[string]string, cdiSpecDirs []string) NewContainerOpts {
+	return func(ctx context.Context, _ *Client, c *containers.Container) error {
+		// TODO: Once CRI is extended with native CDI support this will need to be updated...
+		_, cdiDevices, err := cdi.ParseAnnotations(annotations)
```

Comment:
Does Kubernetes have a standard way to handle annotations to CRI migration? Would we ignore annotations if the corresponding CRI field is used?

Reply:
Thinking out loud.. The injection in kubelet should happen through the cdi, we will need to pass them the CRI version of the actual connection, between kubelet and the CRI runtime, if they don't already have it.  This will tell them if they should inject annotations or new fields.  When that happens they/we can check the CRI Request field(s) first if set that should override annotations and if unset we can check for annotations.  Then we can remove the annotation support when all CRI release API versions being supported have the fields (this because we can be sure cdi will do the right thing). Worst case throw a case into the validator.

Reply:
I don't think Kubernetes has one. At least I don't know about it. We'll have to take care about it when we move from using annotations to using new CRI field. We'll most probably need to support both annotations and CRI field to keep new API(the one that will use CRI field) compatible with the old one. I agree with @mikebrow about the procedure.